### PR TITLE
Adds list of types that shouldn't be saved as output (e.g., FFT plans)

### LINKED ISSUE #231: Bug when trying to saveproblem and params contains an fft plan
This part here errors

https://github.com/FourierFlows/FourierFlows.jl/blob/a453861464c5236695bd84feceddb0f5d3c5290d/src/output.jl#L114

when, e.g., `params` contains an fft plan. This is, e.g., the case at 

https://github.com/FourierFlows/GeophysicalFlows.jl/blob/061cad40a691f3f147e3e6fd6beaa72553d9f83c/src/multilayerqg.jl#L96

### PR DESCRIPTION
**Pull request by navidcy**
Closes #231.

### CODE REVIEW

File: src/output.jl [hunk 1]
```diff
@@ -71,36 +72,41 @@ function savefields(file::JLD2.JLDFile{JLD2.MmapIO}, grid::TwoDGrid)
...
-  nothing
+  
+  return nothing
 end
 
 function savefields(file::JLD2.JLDFile{JLD2.MmapIO}, params::AbstractParams)
   for name in fieldnames(typeof(params))
     field = getfield(params, name)
-    if !(typeof(field) <: Function)
+    if !(typeof(field) <: Function) && !(typeof(field) <: FFTW.FFTWPlan)
```
Comment (glwagner) [inline src/output.jl:90]: You could make this design more sustainable by defining a variable

```
forbidden_output_types = [Function, FFTW.FFTWPlan]
```

and then changing the above line too

```
if !(any([typeof(field) <: forbidden for forbidden in forbidden_output_types]))
    # etc
```

Comment (glwagner) [reply src/output.jl:90]: I might not have gotten the logic right there, but hopefully the idea is clear...

Comment (navidcy) [reply src/output.jl:90]: Very clear and better indeed; will change it.

**Comment by navidcy** [comment]
@glwagner, I also added some test for this.

**Review by glwagner — APPROVED** [review]

**Pull request merged**